doc/extending/extending_theano_gpu.txt

```diff
@@ -0,0 +1,236 @@
+
+.. _extending_theano_gpu:
+
+==============================
+Extending Theano with a GPU Op
+==============================
+
+.. note::
+
+    This covers the :ref:`gpuarray <gpuarray>` back-end for the GPU.
+
+This tutorial covers how to extend Theano with an op that offers a GPU
+implementation.  It assumes you are familiar with how to write new
+Theano ops.  If that is not the case you should probably follow the
+:ref:`extending_theano` and :ref:`extending_theano_c` section before
+continuing on.
+
+Writing a new GPU op can be done in python for some simple tasks, but
+will usually done in C to access the complete API and avoid paying the
+overhead of a python function call.
+
+Dealing With the Context
+========================
+
+One of the major differences with GPU ops is that they require a
+context (a.k.a. device) to execute.  Most of the time you can infer
+the context to run on from your inputs.  There is a way for the user
```

Comment:
There is documentation about `.transfer()`, but that is really just a sidenote here insiting on the point that it is possible that you get arguments that aren't from the same context in your make_node() and you need to handle it.